extensions/control-plane/provision/provision-oauth2/src/main/java/org/eclipse/edc/connector/provision/oauth2/Oauth2SecretToken.java

```diff
@@ -0,0 +1,40 @@
+/*
+ *  Copyright (c) 2022 Bayerische Motoren Werke Aktiengesellschaft (BMW AG)
+ *
+ *  This program and the accompanying materials are made available under the
+ *  terms of the Apache License, Version 2.0 which is available at
+ *  https://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  SPDX-License-Identifier: Apache-2.0
+ *
+ *  Contributors:
+ *       Bayerische Motoren Werke Aktiengesellschaft (BMW AG) - initial API and implementation
+ *
+ */
+
+package org.eclipse.edc.connector.provision.oauth2;
+
+import com.fasterxml.jackson.annotation.JsonProperty;
+import org.eclipse.edc.connector.transfer.spi.types.SecretToken;
+
+/**
+ * Describes an OAuth2 {@link SecretToken}, that's a field containing the token representation.
+ */
+public class Oauth2SecretToken implements SecretToken {
+
+    private final String token;
+
+    public Oauth2SecretToken(@JsonProperty("token") String token) {
+        this.token = token;
+    }
+
+    @Override
+    public long getExpiration() {
+        return 0;
```

Comment:
OAuth Tokens can indeed have an expiry time encoded in the `expires_in` field, at least in the auth code flow. Check [here](https://www.oauth.com/oauth2-servers/access-tokens/access-token-lifetime/) and the [RFC](https://www.rfc-editor.org/rfc/rfc6749#section-4.2.2) for reference. Clients can choose to ignore it, and "wait" for a HTTP 401, but that seems a bit wasteful to me. 
Also, wouldn't we need to store additional information, such as a refresh token?

Reply:
Unfortunately `expires_in` [is noted as recommended but not required](https://www.oauth.com/oauth2-servers/access-tokens/access-token-response/) so I wouldn't rely on that information, same for `refresh_token`, in that case more specific to the server implementation extension could be implemented.

Reply:
sure, but that would mean it's an optional value, with a default of `0`, and maybe a `Map<String, String> additionalFields`. The way it is now, we _can never have_ an expiry, or a refresh token.

Reply:
As long we aren't using them I can't see any use for extracting them, otherwise we could give the impression of actually supporting these features. I stated that in the README.

Reply:
ok, since you explicitly mentioned the client credentials flow, it should be OK. It would  mean that a new token has to be requested every time, which seems legit.